BDArmory/ModuleWeapon.cs

```diff
@@ -1671,11 +1636,124 @@ void Aim()
+                float closestPassSqrDistance = (projectedTarget - closestPass).sqrMagnitude;
+
+                //Debug.Log($"step {simulationSteps}, passDistance: {closestPassSqrDistance}, solution vector {solutionVector}");
+
+                // if close enough return
+                if (closestPassSqrDistance < hitSqrThreshold)
+                    break;
+
+                // if getting further away, target is beyond max ballistic trajectory, raising it more won't help
+                if (closestPassSqrDistance >= prevClosestPass)
+                {
+                    //Debug.Log($"getting further away: {closestPassSqrDistance}, {prevClosestPass}");
+                    if (closestPassSqrDistance < 256) break;
+                    else return null;
+                }
+                prevClosestPass = closestPassSqrDistance;
+
+                // else adjust solutionVector
+                // we actually need the double precision for the angle, that one is completely intentional
+                Vector3 dirToTarget = projectedTarget - simStartPos;
+                Vector3 dirToClosestPass = closestPass - simStartPos;
+                Vector3 sideDirection = Vector3.Cross(dirToTarget, upDir);
+                // yaw
+                //Debug.Log($"adjusting yaw: {(float)Vector3d.Angle(Vector3.ProjectOnPlane(dirToTarget, upDir),Vector3.ProjectOnPlane(dirToClosestPass, upDir))* Mathf.Deg2Rad * Mathf.Sign(Vector3.Dot(dirToClosestPass, sideDirection))} degrees");
+                solutionVector = Quaternion.AngleAxis((float)Vector3d.Angle(
+                        Vector3.ProjectOnPlane(dirToTarget, upDir),
+                        Vector3.ProjectOnPlane(dirToClosestPass, upDir)) 
+                        * Mathf.Sign(Vector3.Dot(dirToClosestPass, sideDirection)),
+                        upDir) * solutionVector;
+                // pitch
+                //Debug.Log($"adjusting pitch {(float)Vector3d.Angle(projectedTarget - simStartPos, Vector3.ProjectOnPlane(closestPass - simStartPos, Vector3.Cross(projectedTarget - simStartPos, upDir)))}");
+                solutionVector = Vector3.RotateTowards(solutionVector, 
+                    upDir * Vector3.Dot(upDir, projectedTarget - closestPass), 
+                    (float)Vector3d.Angle(dirToTarget, 
+                    Vector3.ProjectOnPlane(dirToClosestPass, sideDirection)) * Mathf.Deg2Rad, 0);
+            }
+
+            // looks like we need these things someplace
+            targetLeadDistance = Vector3.Distance(projectedTarget, fireTransforms[0].position);
+            fixedLeadOffset = target - projectedTarget; //for aiming fixed guns to moving target	
+            finalAimTarget = projectedTarget;
+            //Debug.Log($"leadDist: {targetLeadDistance}, leadOffset: {fixedLeadOffset}, aimTarget: {finalAimTarget}");
+            // though I'd say use FiringSolution instead, if feasible
+
+            return solutionVector;
+        }
+
```

Comment:
A method to find a firing solution via trajectory simulation. I hope there's enough comments inside to figure out how it goes. There are also some commented out debug lines, to be deleted later.
Still planning to add an optimization to converge (or give up) quicker when close to the maximum possible distance. This occurs because in that region pitch angle changes produce only minor variation in the achieved distance, thus slowing down convergence.